src/averageAges.js

```diff
@@ -36,8 +44,21 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let result = 0;
+  let ageArray;
+  const arrayWomen = people.filter((woman) => woman.sex === 'f');
+
+  (withChildren)
+    ? ageArray = arrayWomen.filter(woman => {
+      return people.some(child => child.mother === woman.name);
+    })
+      .map(woman => woman.died - woman.born)
+    : ageArray = arrayWomen.map(woman => woman.died - woman.born);
+
+  result = ageArray.reduce((prev, current) => prev + current, 0);
```

Comment:
let us move the logic of looking for average to separate function

Reply:
> let us move the logic of looking for average to separate function

if I understood everything correctly)